packages/graph/hash_graph/lib/graph/src/store/postgres/query/compile.rs

```diff
@@ -321,6 +321,67 @@ impl<'c, 'p: 'c, T: PostgresQueryRecord + 'static> SelectCompiler<'c, 'p, T> {
         }
     }
 
+    fn add_special_relation_conditions(
+        &mut self,
+        relation: Relation,
+        current_alias: Alias,
+        current_table: AliasedTable,
+    ) {
+        match relation {
+            Relation::EntityTypeLinks => {
+                self.artifacts.required_tables.insert(current_table);
+                self.statement
+                    .where_expression
+                    .add_condition(Condition::NotEqual(
+                        Some(Expression::Function(Box::new(Function::JsonExtractPath(
+                            vec![
+                                Expression::Column(
+                                    Column::EntityTypes(EntityTypes::Schema(None))
+                                        .aliased(current_alias),
+                                ),
+                                Expression::Constant(Constant::String("links")),
+                                Expression::Column(
+                                    Column::EntityTypes(EntityTypes::Schema(Some(
+                                        JsonField::Text(&Cow::Borrowed("$id")),
+                                    )))
+                                    .aliased(current_table.alias),
+                                ),
+                            ],
+                        )))),
+                        None,
+                    ));
+            }
+            Relation::EntityTypeInheritance => {
+                self.artifacts.required_tables.insert(current_table);
+                self.statement
+                    .where_expression
+                    .add_condition(Condition::NotEqual(
+                        Some(Expression::Function(Box::new(Function::JsonContains(
+                            Expression::Column(
+                                Column::EntityTypes(EntityTypes::Schema(Some(JsonField::Json(
+                                    &Cow::Borrowed("allOf"),
+                                ))))
+                                .aliased(current_alias),
+                            ),
+                            Expression::Function(Box::new(Function::JsonBuildArray(vec![
+                                Expression::Function(Box::new(Function::JsonBuildObject(vec![(
+                                    Expression::Constant(Constant::String("$ref")),
+                                    Expression::Column(
+                                        Column::EntityTypes(EntityTypes::Schema(Some(
+                                            JsonField::Text(&Cow::Borrowed("$id")),
+                                        )))
+                                        .aliased(current_table.alias),
```

Comment:
I appreciate we are able to use these function calls instead of dealing with funky operator precedence :tada: